ord_schema/interface/query.py

```diff
@@ -25,6 +27,155 @@
+        self.reaction_ids.append(reaction_id)
+
+    def set_reaction_smiles(self, reaction_smiles):
+        self.reaction_smiles = reaction_smiles
+
+    def set_tanimoto_threshold(self, threshold):
+        self.tanimoto_threshold = threshold
+
+    def use_stereochemistry(self):
+        self.do_chiral_ss = True
+
+    def _inputs_predicate(self):
+        exprs = []
+        for pattern, mode in self.inputs:
+            if mode == Predicate.MatchMode.EXACT:
+                expr = f"inputs.smiles = '{pattern}'"
+            elif mode == Predicate.MatchMode.SIMILAR:
+                expr = f"rdki.mfp2%morganbv_fp('{pattern}')"
+            elif mode == Predicate.MatchMode.SUBSTRUCTURE:
+                expr = f"rdki.m@>'{pattern}'"
+            elif mode == Predicate.MatchMode.SMARTS:
+                expr = f"rdki.m@>'{pattern}'::qmol"
+            exprs.append(expr)
+        return ' AND '.join(exprs) or 'TRUE'
+
+    def _output_predicate(self):
+        exprs = []
+        if self.output is not None:
+            pattern, mode = self.output
+            if mode == Predicate.MatchMode.EXACT:
+                expr = f"outputs.smiles = '{pattern}'"
+            elif mode == Predicate.MatchMode.SIMILAR:
+                expr = f"rdko.mfp2%morganbv_fp('{pattern}')"
+            elif mode == Predicate.MatchMode.SUBSTRUCTURE:
+                expr = f"rdko.m@>'{pattern}'"
+            elif mode == Predicate.MatchMode.SMARTS:
+                expr = f"rdko.m@>'{pattern}'::qmol"
+            exprs.append(expr)
+        return ' AND '.join(exprs) or 'TRUE'
+
+    def _reactions_predicate(self):
+        exprs = []
+        if len(self.reaction_ids) > 0:
+            clauses = []
+            for reaction_id in self.reaction_ids:
+                clauses.append(f"reactions.reaction_id='{reaction_id}'")
+            expr = ' OR '.join(clauses)
+            exprs.append(f'({expr})')
+        if self.reaction_smiles is not None:
+            # TODO(kearnes): Correctly express RDKit fingerprint queries.
+            expr = f"reactions.reaction_smiles='{self.reaction_smiles}'"
+            exprs.append(expr)
+        return ' AND '.join(exprs) or 'TRUE'
+
+    def query(self):
+        return f"""
+            SET rdkit.do_chiral_ss={self.do_chiral_ss};
+            SET rdkit.tanimoto_threshold={self.tanimoto_threshold};
+
+            SELECT DISTINCT reactions.serialized
+            FROM reactions
+            INNER JOIN inputs ON inputs.reaction_id = reactions.reaction_id
+            INNER JOIN outputs ON outputs.reaction_id = reactions.reaction_id
+            INNER JOIN rdk.inputs rdki ON rdki.reaction_id = reactions.reaction_id
+            INNER JOIN rdk.outputs rdko ON rdko.reaction_id = reactions.reaction_id
+            WHERE
+                {self._inputs_predicate()}
+                AND {self._output_predicate()}
+                AND {self._reactions_predicate()}
+            LIMIT 100;
+        """
+
+    def query_ids(self):
```

Comment:
This query should just be a simple lookup in the `reactions` table. Probably better to add this to the `OrdPostgres` class.